### PR TITLE
Add CUDA Stubs path to LIBRARY_PATH. This fixes #1463

### LINKED ISSUE #1463: CUDA stubs path missing
In order for the NVML and other libraries to be found, one needs to add 
$CUDA_ROOT/lib64/stubs to $LIBRARY_PATH (not $LD_LIBRARY_PATH)

### PR DESCRIPTION
**Pull request by AndiH**
Also fixes #6602 of easybuilders/easybuild-easyconfigs (https://github.com/easybuilders/easybuild-easyconfigs/issues/6602)

### CODE REVIEW

File: easybuild/easyblocks/c/cuda.py [hunk 1]
```diff
@@ -160,10 +160,14 @@ def make_module_req_guess(self):
...
             lib_path.append('extras/CUPTI/lib64')
             inc_path.append('extras/CUPTI/include')
+            bin_path.append('nvvm/bin')
+            lib_path.append('nvvm/lib64')
+            inc_path.append('nvvm/include')
 
         guesses.update({
             'PATH': bin_path,
             'LD_LIBRARY_PATH': lib_path,
+            'LIBRARY_PATH': ['lib64/stubs'],
```
[the remark this answers is not in the document]
Comment (AndiH) [reply easybuild/easyblocks/c/cuda.py]: `LIBRARY_PATH` wasn't set at all before, so I did not want to change too much. Two notes to regarding your suggestion:

1. `lib_path+['lib64/stubs']`: There's a `lib64` too much!
2. `lib_path`: That would also include `extras/CUPTI/lib64` and `nvvm/lib64` resulting in `lib64/stubs` being added there…

[the remark this answers is not in the document]
Comment (damianam) [reply easybuild/easyblocks/c/cuda.py]: Mmmmh, I see `LIBRARY_PATH` set to `pathJoin(root, "lib64")` in the module, when installing it without this fix. Where did you see it as not set at all? Or did you meant that it was not explicitly updated in `guesses.update`? If the latter, it wasn't explicitly mentioned here, but if you update the dictionary you are overwriting the old definition of that key, which was indeed defined.

Regarding `1.`: I am not sure what you mean :-?
Regarding `2.`: That's correct. Maybe just change it to `'LIBRARY_PATH': ['lib64', 'lib64/stubs'],`?

[the remark this answers is not in the document]
Comment (AndiH) [reply easybuild/easyblocks/c/cuda.py]: I probably don't know enough about EasyBuild. Indeed I thought that `guesses.update()` would be the only definition of `LIBRARY_PATH`.

1. Oh! I was stupied. I forgot that `lib_path` is a list and `+` just appends the list. For some reason I thought of the `+` more as a cross product, adding `lib64/stubs` to every entry of `lib_path`.
2. Ok, fine with me!

**Comment by akesandgren** [comment]
You should use os.join.path() instead of explicitly using "a/b"
That's all as far as I can see.

**Comment by boegel** [comment]
I'll let @akesandgren's remark slip, since the added lines just follow the style of the preceding ones.

@damianam Can you look into cleaning that up in your follow-up PR?

**Comment by mboisson** [comment]
@boegel, sorry to do thread necropsy, but just as a note, adding stubs to `LIBRARY_PATH` is causing us problems, because it ends up in our binaries `RUNPATH`. 

If this happens and you don't use `LD_LIBRARY_PATH`  (because you are using `RUNPATH`), you will have problems at runtime. 

This is probably something you need to check when you are using runpath/rpath with easybuild  (for Compute Canada, @bartoldeman will have to adjust our linker wrapper or we need to not put the stubs in our `LIBRARY_PATH`) 

```
A stub object is a shared object, built entirely from mapfiles, that supplies the same linking interface as the real object, while containing no code or data. Stub objects cannot be used at runtime.
```

**Comment by bartoldeman** [comment]
I adjusted our linker wrapper to not convert -L.../stubs into -rpath .../stubs now.

I guess the same could be done to https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/scripts/rpath_args.py which does similar things but I haven't tested EB's rpath support yet.

**Comment by boegel** [comment]
@mboisson @bartoldeman Please open an issue in framework to get this fixed in our RPATH support, indeed in `rpath_args.py`

**Comment by mboisson** [comment]
done

**Pull request merged**